tests/test_ir.py

```diff
@@ -149,6 +167,89 @@ def test_op_clone_with_regions():
     assert if2.false_region.op is not if_.false_region.op
 
 
+@irdl_op_definition
+class SuccessorOp(IRDLOperation):
+    """
+    Utility operation that requires a successor and has the IsTerminator
+    trait.
+    """
+
+    name = "test.successor_op"
+
+    successor: Successor = successor_def()
+
+    traits = frozenset([IsTerminator()])
+
+
+def test_block_branching_to_another_region_wrong():
+    """
+    Tests that an operation cannot have successors that branch to blocks of
+    another region.
+    """
+    block1 = Block([TestOp.create(), TestOp.create()])
+    region1 = Region([block1])
+
+    op0 = TestOp.create(successors=[block1])
+    block0 = Block([op0])
+    region0 = Region([block0])
+    region0 = TestOp.create(regions=[region0, region1])
+
+    outer_block = Block([region0])
+
+    with pytest.raises(
+        VerifyException,
+        match="Branching to a block of a different region",
+    ):
+        outer_block.verify()
+
+
+def test_block_not_branching_to_another_region():
+    """
+    Tests that an operation can have successors that branch to blocks of the
+    same region.
+    """
+    block0 = Block()
+
+    op0 = SuccessorOp.create(successors=[block0])
+    block1 = Block([op0])
+
+    _ = Region([block0, block1])
+
+    op0.verify()
+
+
+def test_empty_block_with_no_parent_region_requires_no_terminator():
+    """
+    Tests that an empty block belonging no parent region requires no terminator
+    operation.
+    """
+    block0 = Block([])
+
+    block0.verify()
+
```

Comment:
Out of curiosity; this should later be replaced by the NoTerminator trait, right?

Reply:
@PapyChacal I added a test with a TODO that should fail once the `NoTerminator` trait is added in.

Reply:
Lovely, no worries for the confusion, I think that's what I hoped for!
Thank you for working on this!!